x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/flattened/FlattenedFeatureSetUsage.java

```diff
@@ -6,19 +6,57 @@
 
 package org.elasticsearch.xpack.core.flattened;
 
+import org.elasticsearch.Version;
 import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.XContentBuilder;
 import org.elasticsearch.xpack.core.XPackFeatureSet;
 import org.elasticsearch.xpack.core.XPackField;
 
 import java.io.IOException;
+import java.util.Objects;
 
 public class FlattenedFeatureSetUsage extends XPackFeatureSet.Usage {
+    private final int fieldCount;
 
     public FlattenedFeatureSetUsage(StreamInput input) throws IOException {
         super(input);
+        this.fieldCount = input.getVersion().onOrAfter(Version.V_8_0_0) ? input.readInt() : 0;
```

Comment:
I assume for 7.x the version would be changed to `V_7_6_0` here and in `writeTo`.

Reply:
Right, that was my plan. Then I'll make an update on master to change to `V_7_6_0` as well.